luigi/hdfs.py

```diff
@@ -105,6 +105,51 @@ def list_path(path):
         return [path, ]
     return [str(path), ]
 
+def is_dangerous_rm_path(path):
+    """ Finds out if the path is too dangerous to remove.
+    * empty path
+    * / root path
+    * one level deep root path, i.e. /etc or /opt
+    :return boolean, True if too dangerous! Relative path is always ok
+
+    >>> is_dangerous_rm_path(None)
+    True
+    >>> is_dangerous_rm_path('')
+    True
+    >>> is_dangerous_rm_path(' ')
+    True
+    >>> is_dangerous_rm_path('/opt')
+    True
+    >>> try:
+    ...     is_dangerous_rm_path('//')
+    ... except ValueError:
+    ...     pass
+    >>> is_dangerous_rm_path('/opt/')
+    True
+    >>> is_dangerous_rm_path('/opt/foo/bar')
+    False
+    >>> is_dangerous_rm_path('my_relative')
+    False
+    """
+    if not path:
+        return True
+
+    if '//' in path:
+        raise ValueError('Funny path, double slash: %s' % path)
```

Comment:
You mean the '//' check? Somewhat a kludge to simplify the following code.
Also, since ultimately this path string might be passed to a hadoop fs -rmr process, I would definitely be concerned at the behavior of:
   hadoop fs -rmr //

Not sure if the hadoop CLI would try to delete the root directory, or exit 1.